src/data-source/DataSource.ts

```diff
@@ -641,6 +641,20 @@ export class DataSource {
                     )
                 }
             }
+            if (
+                target !== null &&
+                typeof target === "function" &&
+                typeof target.name === "string"
+            ) {
+                if (target.name.indexOf(".") !== -1) {
+                    return metadata.tablePath === target.name
+                } else {
+                    return (
+                        metadata.name === target.name ||
+                        metadata.tableName === target.name
+                    )
+                }
```

Comment:
I figured this logic was in here for a specific use case looking at the `isObject` logic? Let me know if this would need to be changed?

Unsure how a class/function could have a `.` in its name though.